sql/expression/case.go

```diff
@@ -190,8 +192,10 @@ func (c *Case) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 		if err != nil {
 			return nil, err
 		}
-		ret, _, err := t.Convert(val)
-		return ret, err
+		if ret, _, err := t.Convert(val); err == nil {
```

Comment:
not sure i understand, we were failing to convert a decimal to an int8 or something?

Reply:
we were incorrectly returning an `Out of bounds value for decimal type` error